rotkehlchen/exchanges/kraken.py

```diff
@@ -781,3 +800,219 @@ def query_online_income_loss_expense(
             end_ts: Timestamp,  # pylint: disable=unused-argument
     ) -> List[LedgerAction]:
         return []  # noop for kraken
+
+    def query_pairs(self) -> None:
+        self.pairs = set(self.api_query('AssetPairs').keys())
+
+    def kraken_asset_to_pair(self, asset1: Asset, asset2: Asset) -> Optional[str]:
+        """Given a pair of assets find the kraken pair that has this two assets as
+        base and quote"""
+        asset1_symbol, asset2_symbol = asset1.symbol, asset2.symbol
+        asset1_kraken_tr = WORLD_TO_KRAKEN.get(asset1_symbol, asset1_symbol)
+        asset2_kraken_tr = WORLD_TO_KRAKEN.get(asset2_symbol, asset2_symbol)
+
+        for a, b in permutations(
+            [
+                ('a', asset1_symbol),
+                ('b', asset2_symbol),
+                ('a', asset1_kraken_tr),
+                ('b', asset2_kraken_tr),
+            ],
+            2,
+        ):
+            if a[0] == b[0]:
+                continue
+            pair_candidate = a[1] + b[1]
+            if pair_candidate in self.pairs:
+                return pair_candidate
+        return None
+
+    def process_kraken_trades(
+        self,
+        raw_data: List[HistoryEvent],
+    ) -> List[Trade]:
+        """
+        Given a list of history events we process them to create Trade objects. The valid
+        History events type are
+        - Trade
+        - Receive
+        - Spend
+
+        A pair of receive and spend events can be a trade and kraken uses this kind of event
+        for instant trades and trades made from the phone app. What we do is verify that is a trade
+        is to check if we can find a pair with the same event id.
+
+        May raise:
+        - RemoteError if the paris couldn't be correctly queried
+        """
+        if len(self.pairs) == 0:
+            self.query_pairs()
```

Comment:
you should move this if logic inside the function. And rename it to `query_pairs_if_needed()`. Otherwise the function is pretty much an empty shell.

```suggestion
        if len(self.pairs) == 0:
            self.query_pairs()
```